jobs/aws_cpi/spec

```diff
@@ -60,6 +64,14 @@ properties:
     description: Encrypts all instances' volumes with the given KMS key. (aws.encrypted) should be true
     example: arn:aws:kms:us-east-1:XXXXXX:key/e1c1f008-779b-4ebe-8116-0a34b77747dd
     default: null
+  aws.extend_ebs_volume.wait_time_factor:
+    description: |
+      Default wait time for AWS operations is ≈25 minutes. When growing
+      persistent disk size above 4TB, this maximum wait time may need to be
+      increased. The value here is a factor of the base ≈25 minutes wait
+      time. For example a factor of 3 would lead to a wait time of ≈75
+      minutes.
+    default: 4
```

Comment:
It feels unfortunate that this `ResourceWait` model is leaking out to the operator configuration. It seems odd from an operator perspective to be multiplying `25 * wait_time_factor` to configure a timeout.

Reply:
Yeah, the watch/retry/wait logic with increasing delays between retries is a bit odd when only waiting for some resource state to become some expected state.

In this work, I've tried to bring minimal yet relevant changes to address the need for longer wait times when it comes to resizing disks. I think we're pointing out here a need for the CPI to implement a different algorithm for waiting long AWS operations to finish. But such refactoring could not reasonably be part of this patch here.

Reply:
We're going to take a look at it today

Reply:
Yeah, it seems like `ResourceWait` is unfortunately too complex to take any sane "time limit" inputs. It would be much better if `ResourceWait` were centered around "maximum time" rather than "maximum tries".

The "maximum tries" focus of `ResourceWait` makes it so the only way to take a "time limit" input, would be to then calculate back to the number of tries to make it true, which would be some pretty ugly code.

It would be great to refactor all of `ResourceWait` to be more sane, but that feels like a task for another day.